File: pytools/examples/php/client/protocol/AckChatSendOk.php
<?php
namespace protocol;

use network\Packet;


class AckChatSendOk
{
	private $channel = 0;
	private $uid = 0;
	private $uname = '';
	private $content = '';


	public function __construct(&$packet)
	{
		$this->decode($packet);
	}

	public function decode(&$packet)
	{
		$this->channel = $packet->readU8();
		$this->uid = $packet->readU32();
		$this->uname = $packet->readString();
		$this->content = $packet->readString();
	}


	public function getChannel()
	{
		return $this->channel;
	}

	public function getUid()
	{
		return $this->uid;
	}

	public function getUname()
	{
		return $this->uname;
	}

	public function getContent()
	{
		return $this->content;
	}

}

File: pytools/data/data.code.lua.client/AckChatSendOk.lua
module("AckChatSendOk", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = AckChatSendOk})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.channel = 0
	self.uid = 0
	self.uname = 0
	self.content = 0
end


function decode(self, pack)
	self.channel = pack:readU8()
	self.uid = pack:readU32()
	self.uname = pack:readString()
	self.content = pack:readString()
	return self
end


function getChannel(self)
	return self.channel
end

function getUid(self)
	return self.uid
end

function getUname(self)
	return self.uname
end

function getContent(self)
	return self.content
end

